pipeline/builder/builder.go

```diff
@@ -203,6 +203,24 @@ func (b *Builder) buildDeployStage(index int, s config.Stage) (*types.DeployStag
 			}
 		}
 
+		var deploymentSettings *types.Deployment
+
+		if deployment := group.Deployment; deployment != nil {
+
+			deploymentSettings = &types.Deployment{
+				Enabled:         deployment.Enabled,
+				MinReadySeconds: deployment.MinReadySeconds,
+				DeploymentStrategy: types.DeploymentStrategy{
+					RollingUpdate: types.RollingUpdate{
+						MaxSurge:       fmt.Sprintf("%d", deployment.DeploymentStrategy.RollingUpdate.MaxSurge),
+						MaxUnavailable: fmt.Sprintf("%d", deployment.DeploymentStrategy.RollingUpdate.MaxUnavailable),
+					},
+					Type: "RollingUpdate",
+				},
+				RevisionHistoryLimit: deployment.RevisionHistoryLimit,
+			}
```

Comment:
If you have this, do the maxASGs and such need to be "erased" from the struct? Things like "redblack" become irrelevant right?

Reply:
it seems as if spinnaker holds onto that in the json. I made this change manually from rs -> deploys and noted that the json only _adds_ to it. see diff here: http://www.mergely.com/TTOziUMa/

Reply:
actually I tested this in spinnaker and it isn't performing as intended. it will scale up a deploy, then scale down the same deploy. will get back to you on this